helper/common/common_test.go

```diff
@@ -168,3 +169,80 @@ func Test_SafeAddUint64(t *testing.T) {
 		})
 	}
 }
+
+func TestNewUnsafePool(t *testing.T) {
+	pool := NewUnsafePool[int]()
+
+	if pool == nil {
+		t.Errorf("NewUnsafePool returned nil")
```

Comment:
Remove conditions in tests and just use `require`:

```suggestion
		require.NotNilf(t, pool, "NewUnsafePool returned nil")
```